From 5a1c6339d7745122deb5dc90e17950e79013fbbd Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Tue, 27 Jun 2023 10:22:01 +0200
Subject: [PATCH] use public API to clear hypre errors

instead of setting error flag, which has changed type in 2.29 to a struct
---
 src/ksp/pc/impls/hypre/hypre.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ksp/pc/impls/hypre/hypre.c b/src/ksp/pc/impls/hypre/hypre.c
index 5faa109d207..ab379f3d6a6 100644
--- a/src/ksp/pc/impls/hypre/hypre.c
+++ b/src/ksp/pc/impls/hypre/hypre.c
@@ -443,7 +443,7 @@ static PetscErrorCode PCApply_HYPRE(PC pc, Vec b, Vec x)
       HYPRE_Int hierr = (*jac->solve)(jac->hsolver, hmat, jbv, jxv);
       if (hierr) {
         PetscCheck(hierr == HYPRE_ERROR_CONV, PETSC_COMM_SELF, PETSC_ERR_LIB, "Error in HYPRE solver, error code %d", (int)hierr);
-        hypre__global_error = 0;
+        HYPRE_ClearAllErrors();
       }
     } while (0));
 
@@ -633,7 +633,7 @@ static PetscErrorCode PCApplyTranspose_HYPRE_BoomerAMG(PC pc, Vec b, Vec x)
       if (hierr) {
         /* error code of 1 in BoomerAMG merely means convergence not achieved */
         PetscCheck(hierr == 1, PETSC_COMM_SELF, PETSC_ERR_LIB, "Error in HYPRE solver, error code %d", (int)hierr);
-        hypre__global_error = 0;
+        HYPRE_ClearAllErrors();
       }
     } while (0));
 
-- 
2.41.0

